python/amici/sbml_import.py

```diff
@@ -1402,42 +1219,48 @@ def _replace_in_all_expressions(self,
             for k in d:
                 d[k] = d[k].subs(old, new)
 
-        for symbol in ['species', 'observable', 'llhy', 'sigmay']:
+        for symbol in [SymbolId.SPECIES, SymbolId.OBSERVABLE, SymbolId.LLHY,
+                       SymbolId.SIGMAY]:
             if not self.symbols.get(symbol, None):
                 continue
-            self.symbols[symbol]['value'] = \
-                self.symbols[symbol]['value'].subs(old, new)
+            for element in self.symbols[symbol].values():
+                if symbol is SymbolId.SPECIES:
+                    repl = self._make_initial(new)
+                else:
+                    repl = new
+
+                element['value'] = element['value'].subs(old, repl)
```

Comment:
Is `self.symbols[THING]['value']` always the initial value of `THING`? If so, might be more readable as `'init'` instead of `'value'`.

Reply:
It is only for species, for everything else it is the symbolic expression that defines the value.

Reply:
but I agree, it probably makes sense to change this for species